src/main/java/com/amazon/opendistroforelasticsearch/knn/index/v1736/KNNIndex.java

```diff
@@ -61,25 +60,38 @@ public long getIndexSize() {
         return this.indexSize;
     }
 
-    /**
-     * determines the size of the hnsw index on disk
-     * @param indexPath absolute path of the index
-     *
-     */
-    public void computeFileSize(String indexPath) {
-        if (indexPath != null && !indexPath.isEmpty()) {
-            File file = new File(indexPath);
-            if (!file.exists() || !file.isFile()) {
-                setIndexSize(0);
-            } else {
-                setIndexSize(file.length()/1024 + 1); // convert to KB and round up
+    public KNNQueryResult[] queryIndex(final float[] query, final int k, final String[] algoParams) throws IOException {
+        Lock readLock = readWriteLock.readLock();
+        readLock.lock();
+
+        try {
+            if (this.isClosed) {
+                throw new IOException("Index is already closed");
```

Comment:
Per our call, the check as to whether the index has been closed needs to happen after acquisition of the read lock, which means that the write lock is not actively held. Without that, isClosed could be false, but then upon acquisition of the read lock it could've flipped to true.

Reply:
https://github.com/opendistro-for-elasticsearch/k-NN/pull/24/commits/112b83c38ab254f86b20c737ca403021ed001df6 fixes issue I mentioned on the call where write lock wasn't actually being acquired